### PR TITLE
Feature/refactor date input and duration to remove duplicated code

### LINKED ISSUE #2190: Duration and date input components use much of the same code in each macro
The duration and date input component contains much of the same logic duplicated across the two macros. In some areas the overall logic is the same but the coding patterns are inconsistent.

We should refactor these components to possibly use an "intermediate" macro that handles most of the logic such as mutually exclusive, fieldset, length logic etc

### PR DESCRIPTION
**Pull request by precious-onyenaucheya-ons**
### What is the context of this PR?
fixes #2190 
Refactored and improved the date input and duration components by consolidating common code into a new macro called "date input field." This macro now manages shared logic, including mutually exclusive, fieldset, and input logic.

### How to review this PR
Check that all logic remains the same and all the tests pass 

### Checklist

This needs to be completed by the person raising the PR.

- [x] I have selected the correct Assignee
- [x] I have linked the correct Issue

### CODE REVIEW

**Review by SriHV — APPROVED** [review]

File: src/components/date-field-input/_macro.njk [hunk 1]
```diff
@@ -0,0 +1,87 @@
+{% from "components/mutually-exclusive/_macro.njk" import onsMutuallyExclusive %}
+{% from "components/fieldset/_macro.njk" import onsFieldset %}
+{% from "components/field/_macro.njk" import onsField %}
```
Comment (rmccar) [inline src/components/date-field-input/_macro.njk:3]: This import isn't being used

File: src/components/date-field-input/_macro.njk [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                "value": field.value,
+                "min": field.min,
+                "max": field.max,
+                "maxLength": field.maxLength,
+                "suffix": field.suffix,
+                "label": field.label,
+                "attributes": field.attributes,
+                "error": field.error,
+                "fieldId": field.fieldId
+                })
```
Comment (rmccar) [inline src/components/date-field-input/_macro.njk:26]:
```suggestion
            {{
                onsInput({
                    "id": field.id,
                    "classes": field.classes,
                    "width": field.width,
                    "type": "number",
                    "name": field.name,
                    "value": field.value,
                    "min": field.min,
                    "max": field.max,
                    "maxLength": field.maxLength,
                    "suffix": field.suffix,
                    "label": field.label,
                    "attributes": field.attributes,
                    "error": field.error,
                    "fieldId": field.fieldId
                })
```

File: src/components/duration/_macro.njk [hunk 3]
```diff
@@ -15,85 +10,65 @@
...
+        "mutuallyExclusive": params.mutuallyExclusive,
+        "numberOfFields": numberOfFields,
+        "fields": fields,
+        "error": params.error,
+        "legend": params.legendOrLabel,
+        "dontWrap": params.dontWrap,
+        "classes": params.classes,
+        "legendClasses": params.legendClasses,
+        "legendIsQuestionTitle": params.legendIsQuestionTitle
+        }) }}
```
Comment (rmccar) [inline src/components/duration/_macro.njk:73]:
```suggestion
    }) }}
```

**Review by rmccar — COMMENTED** [review]
Quite a few code formatting issues. Theres quite a few lines I havent commented on that have trailing whitespace. I know it might seem nit picky but as we dont have a linter at the moment for nunjucks it is important to keep on top of this stuff to keep our code clean and readable. 
There doesn't seem to be a section for reviewing instructions in the description either.

**Review by rmccar — APPROVED** [review]

**Pull request merged**